.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseTimePeriodParser.cs

```diff
@@ -137,15 +137,20 @@ private DateTimeResolutionResult ParseSimpleCases(string text, DateObject refere
 
                     if (!string.IsNullOrEmpty(amStr) || rightAmValid)
                     {
-                        
-                        if (beginHour >= 12)
+                        if (endHour >= 12)
+                        {
+                            endHour -= 12;
+                        }
+
+                        if (beginHour >= 12 && beginHour - 12 < endHour)
                         {
                             beginHour -= 12;
                         }
 
-                        if (endHour >= 12)
+                        // Resolve case like "11 to 3am"
+                        if (beginHour < 12 && beginHour > endHour)
```

Comment:
It is better that the PR title include the Issue number. Then I can check the details of the bug

Reply:
In the description, not in the title. :)